core/src/main/scala/com/databricks/labs/remorph/intermediate/expressions.scala

```diff
@@ -161,7 +161,25 @@ case class Cast(
     type_str: String = "",
     returnNullOnError: Boolean = false,
     timeZoneId: Option[String] = None)
-    extends Unary(expr)
+    extends Unary(expr) {
+
+  override def makeCopy(newArgs: Array[AnyRef]): Expression = {
```

Comment:
I don't understand why we need overriding `makeCopy` here, is there a test somewhere showcasing the problem/solution ?

Reply:
It crashes if you don't. `TreeNode.makeCopy` docs say:
```
Must be overridden by child classes that have
   * constructor arguments that are not present in the productIterator.

```
I don't know exactly why only 1 arg of the `Cast` ctor is seen by the `productIterator` (maybe because they are not `val`s), but it's what happens.
This is tested via the test cases in `resources/functional/snowflake/core_engine/lca`